src/access/AccessRestriction.h

```diff
@@ -0,0 +1,288 @@
+     *
+     * If access is not restricted, CHIP_NO_ERROR will be returned.
+     *
+     * @retval #CHIP_ERROR_ACCESS_DENIED if access is denied.
+     * @retval other errors should also be treated as restricted/denied.
+     * @retval #CHIP_NO_ERROR if access is not restricted/denied.
+     */
+    CHIP_ERROR Check(const SubjectDescriptor & subjectDescriptor, const RequestPath & requestPath);
+
+    /**
+     * Request a review of the access restrictions for a fabric.
+     *
+     * @param [in]  fabricIndex  The index of the fabric requesting a review.
+     * @param [in]  arl          An optinal list of access restriction entries to review.  If null, all entries will be reviewed.
+     * @param [out] token        The unique token for the review, which can be matched to a review update event.
+     */
+    CHIP_ERROR RequestFabricRestrictionReview(FabricIndex fabricIndex, const std::vector<Entry> & arl, uint64_t & token)
+    {
+        token = ++mNextToken;
+        return DoRequestFabricRestrictionReview(fabricIndex, token, arl);
+    }
+
+    using EntryIterator = std::vector<SharedPtr<Entry>>::iterator;
+
+    /**
+     * Get iterator over the commissioning entries, which are used during commissioning
+     * and also installed as defaults for a fabric upon completion of commissioning.
+     *
+     * @param [out] begin iterator pointing to the beginning of the entries
+     * @param [out] end iterator pointing to the end of the entries
+     */
+    static CHIP_ERROR CommissioningEntries(EntryIterator & begin, EntryIterator & end)
+    {
+        begin = sCommissioningEntries.begin();
+        end   = sCommissioningEntries.end();
+
+        return CHIP_NO_ERROR;
+    }
+
+    /**
+     * Get iterator over entries in the access restriction list by fabric.
+     *
+     * @param [in]  fabricIndex Iteration is confined to fabric
+     * @param [out] begin iterator pointing to the beginning of the entries
+     * @param [out] end iterator pointing to the end of the entries
+     */
+    CHIP_ERROR Entries(const FabricIndex fabricIndex, EntryIterator & begin, EntryIterator & end)
+    {
+        if (mFabricEntries.find(fabricIndex) == mFabricEntries.end())
+        {
+            return CHIP_ERROR_NOT_FOUND;
+        }
+
+        begin = mFabricEntries[fabricIndex].begin();
+        end   = mFabricEntries[fabricIndex].end();
+
+        return CHIP_NO_ERROR;
+    }
+
+    /**
+     * Add a restriction entry to the commissioning access restriction list. This list is automatically
+     * applied to fabrics upon completion of commissioning. This request will fail if there is already
+     * an entry with the same cluster and endpoint.
+     *
+     * @param [in]  entry       The entry to add to the commissioning access restriction list.
+     * @return                  CHIP_NO_ERROR if the entry was successfully created, or an error code.
+     */
+    static CHIP_ERROR CreateCommissioningEntry(SharedPtr<Entry> entry);
```

Comment:
why is the API different for CreateCommissioningEntry and CreateEntry (aside from the fabricIndex)

Reply:
This data management model has been greatly simplified and this isnt an issue anymore.